wp-multi-network/includes/classes/class-wp-ms-networks-api.php

```diff
@@ -0,0 +1,859 @@
+			'self' => array(
+				'href' => rest_url( sprintf( '%s/%s/%d', $this->namespace, $this->rest_base, $network->id ) ),
+			),
+			'collection' => array(
+				'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param string $query_param Query parameter.
+	 * @return string The normalized query parameter.
+	 */
+	protected function normalize_query_param( $query_param ) {
+		switch ( $query_param ) {
+			case 'include':
+				$normalized = 'network__in';
+				break;
+			default:
+				$normalized = $query_param;
+				break;
+		}
+
+		return $normalized;
+	}
+
+	/**
+	 * Prepares a single network to be inserted into the database.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return array|WP_Error Prepared network, otherwise WP_Error object.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_network = array();
+
+		if ( isset( $request['path'] ) ) {
+			$prepared_network['path'] = $request['path'];
+		}
+
+		if ( isset( $request['domain'] ) ) {
+			$prepared_network['domain'] = $request['domain'];
+		}
+
+		/**
+		 * Filters a network after it is prepared for the database.
+		 *
+		 * Allows modification of the network right after it is prepared for the database.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param array           $prepared_network The prepared network data for `insert_network`.
+		 * @param WP_REST_Request $request          The current request.
+		 */
+		return apply_filters( 'rest_preprocess_network', $prepared_network, $request );
+	}
+
+	/**
+	 * Retrieves the network's schema, conforming to JSON Schema.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'              => 'http://json-schema.org/draft-04/schema#',
+			'title'                => 'network',
+			'type'                 => 'object',
+			'properties'           => array(
+				'id'               => array(
+					'description'  => __( 'Unique identifier for the object.', 'wp-multi-network' ),
+					'type'         => 'integer',
+					'context'      => array( 'view', 'edit', 'embed' ),
+					'readonly'     => true,
+				),
+				'domain'           => array(
+					'description'  => __( 'The domain of the network object.', 'wp-multi-network' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit', 'embed' ),
+				),
+				'path'           => array(
+					'description'  => __( 'The path of the network object', 'wp-multi-network' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit', 'embed' ),
+				),
+				'link'             => array(
+					'description'  => __( 'URL to the object.', 'wp-multi-network' ),
+					'type'         => 'string',
+					'format'       => 'uri',
+					'context'      => array( 'view', 'edit', 'embed' ),
+					'readonly'     => true,
+				),
+			),
+		);
```

Comment:
Let's keep it simple and remove `link` for now. If we want a fourth field, it should be `site_id` for the main site. It would need to have a key `readonly => true` in the schema since you cannot update it through the controller.